### ISSUE TITLE
Double slash in URLs still breaks route matching with variable parts in vapor 4.92.0

### ISSUE THREAD

**Issue by weissi** [vapor/vapor#3142]
Consider this route

```
app.get(":variable", "fixed") { request in
    return "ok \(#line)"
}
```

this should work fine for `http://127.0.0.1:8080/testTeam/fixed` as well as `http://127.0.0.1:8080//testTeam/fixed`. Unfortunately, it doesn't work for `http://127.0.0.1:8080//testTeam/fixed`.

Actual:

```
$ curl http://127.0.0.1:8080/testTeam/fixed
ok 82
$ curl http://127.0.0.1:8080/testTeam//fixed
ok 82
$ curl http://127.0.0.1:8080//testTeam/fixed
{"reason":"Not Found","error":true}
$ curl http://127.0.0.1:8080//testTeam//fixed
{"error":true,"reason":"Not Found"}
```

Expected:

all printing `ok 82`.

Version:

vapor 4.92.0

**Comment by gwynne** [comment]
@weissi Can you show the output when invoking `curl` with `-v` so I can see what actual request is being sent?

**Comment by vlm** [comment]
@gwynne 

```
> curl -v http://127.0.0.1:8080//testTeam//fixed
*   Trying 127.0.0.1:8080...
* Connected to 127.0.0.1 (127.0.0.1) port 8080
> GET //testTeam//fixed HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/8.4.0
> Accept: */*
```
